### PR TITLE
fix: allow CopyObject/PutObjecTags on pre-existing content

### LINKED ISSUE #10475: [resolved] [FS] CopyObject returns 404 Not Found even though headObject succeeds in
When trying to update a objects metadata using copyObject and metadata directive `REPLACE` the call fails with 404 Not Found even though the key exists and a previous headObject returns 200 Ok.

## Expected Behavior
If headObject on a key succeeds it should be possible to call copyObject immediately after.

## Current Behavior
The headObject operation succeeds with 200 Ok but a following copyObject call with the same bucket and key and same source and destination fails with 404 Not Found.

## Steps to Reproduce (for bugs)

1. docker run lallinger/minio-test

Output from the container:
```
Attempting encryption of all config, IAM users and policies on MinIO backend
Endpoint:  http://172.17.0.2:9000  http://127.0.0.1:9000

Browser Access:
   http://172.17.0.2:9000  http://127.0.0.1:9000

Object API (Amazon S3 compatible):
   Go:         https://docs.min.io/docs/golang-client-quickstart-guide
   Java:       https://docs.min.io/docs/java-client-quickstart-guide
   Python:     https://docs.min.io/docs/python-client-quickstart-guide
   JavaScript: https://docs.min.io/docs/javascript-client-quickstart-guide
   .NET:       https://docs.min.io/docs/dotnet-client-quickstart-guide
Added `minio` successfully.
total 40992
-rw-r--r-- 1 root root 41943040 Sep 14 11:40 big
-rw-r--r-- 1 root root    32768 Sep 14 11:40 small
mc: <DEBUG> GET /minio/admin/v3/trace?all=false&err=false HTTP/1.1
Host: localhost:9000
User-Agent: MinIO (linux; amd64) madmin-go/0.0.1 mc/2020-09-03T00:08:28Z
Authorization: AWS4-HMAC-SHA256 Credential=admin/20200914//s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20200914T114039Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Transfer-Encoding: chunked
Cache-Control: no-cache
Content-Security-Policy: block-all-mixed-content
Content-Type: text/event-stream
Date: Mon, 14 Sep 2020 11:40:39 GMT
Vary: Origin
X-Accel-Buffering: no
X-Amz-Request-Id: 1634A397B4643257
X-Xss-Protection: 1; mode=block

mc: <DEBUG> Response Time:  24.610747ms

localhost [REQUEST s3.GetBucketLocation] 11:40:39.608
localhost GET /test/?location=
localhost Proto: HTTP/1.1
localhost Host: localhost:9000
localhost User-Agent: MinIO (linux; amd64) minio-go/v7.0.6 mc/2020-09-03T00:08:28Z
localhost X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
localhost X-Amz-Date: 20200914T114039Z
localhost Authorization: AWS4-HMAC-SHA256 Credential=admin/20200914/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=881c6262d5034bc3639b9ad0f5d2e816efa82bfaaf0f314bbc76ec1ebd91e2c1
localhost Content-Length: 0
localhost
localhost [RESPONSE] [11:40:39.608] [ Duration 173µs  ↑ 77 B  ↓ 378 B ]
localhost 200 OK
localhost Content-Type: application/xml
localhost Server: MinIO/RELEASE.2020-09-10T22-02-45Z
localhost Vary: Origin
localhost X-Amz-Request-Id: 1634A397B5C21D34
localhost X-Xss-Protection: 1; mode=block
localhost Accept-Ranges: bytes
localhost Content-Length: 128
localhost Content-Security-Policy: block-all-mixed-content
localhost <?xml version="1.0" encoding="UTF-8"?>
<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/"></LocationConstraint>
localhost
localhost [REQUEST s3.HeadBucket] 11:40:39.609
localhost HEAD /test/
localhost Proto: HTTP/1.1
localhost Host: localhost:9000
localhost X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
localhost X-Amz-Date: 20200914T114039Z
localhost Authorization: AWS4-HMAC-SHA256 Credential=admin/20200914/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=b778d08c01da3c0b18e20abb8bdb60596561fde87a8aa54425109747b6fb65d9
localhost Content-Length: 0
localhost User-Agent: MinIO (linux; amd64) minio-go/v7.0.6 mc/2020-09-03T00:08:28Z
localhost
localhost [RESPONSE] [11:40:39.609] [ Duration 127µs  ↑ 77 B  ↓ 218 B ]
localhost 200 OK
localhost Accept-Ranges: bytes
localhost Content-Length: 0
localhost Content-Security-Policy: block-all-mixed-content
localhost Server: MinIO/RELEASE.2020-09-10T22-02-45Z
localhost Vary: Origin
localhost X-Amz-Request-Id: 1634A397B5D3D54C
localhost X-Xss-Protection: 1; mode=block
localhost
localhost
localhost [REQUEST s3.ListObjectsV2] 11:40:39.610
localhost GET /test/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=
localhost Proto: HTTP/1.1
localhost Host: localhost:9000
localhost X-Amz-Date: 20200914T114039Z
localhost Authorization: AWS4-HMAC-SHA256 Credential=admin/20200914/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=f95892907ece133f0d9588c70e1d8c6a9f25b05ea44a2ac2e30bc9a7470abb3f
localhost Content-Length: 0
localhost User-Agent: MinIO (linux; amd64) minio-go/v7.0.6 mc/2020-09-03T00:08:28Z
localhost X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
localhost
localhost [RESPONSE] [11:40:39.610] [ Duration 396µs  ↑ 77 B  ↓ 1.1 KiB ]
localhost 200 OK
localhost Content-Length: 926
localhost Content-Security-Policy: block-all-mixed-content
localhost Content-Type: application/xml
localhost Server: MinIO/RELEASE.2020-09-10T22-02-45Z
localhost Vary: Origin
localhost X-Amz-Request-Id: 1634A397B5DE1537
localhost X-Xss-Protection: 1; mode=block
localhost Accept-Ranges: bytes
localhost <?xml version="1.0" encoding="UTF-8"?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Name>test</Name><Prefix></Prefix><KeyCount>2</KeyCount><MaxKeys>1000</MaxKeys><Delimiter>/</Delimiter><IsTruncated>false</IsTruncated><Contents><Key>big</Key><LastModified>2020-09-14T11:40:39.592Z</LastModified><ETag>&#34;00000000000000000000000000000000-1&#34;</ETag><Size>41943040</Size><Owner><ID>02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4</ID><DisplayName></DisplayName></Owner><StorageClass>STANDARD</StorageClass></Contents><Contents><Key>small</Key><LastModified>2020-09-14T11:40:39.569Z</LastModified><ETag>&#34;00000000000000000000000000000000-1&#34;</ETag><Size>32768</Size><Owner><ID>02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4</ID><DisplayName></DisplayName></Owner><StorageClass>STANDARD</StorageClass></Contents><EncodingType>url</EncodingType></ListBucketResult>
localhost
[2020-09-14 11:40:39 UTC]   40MiB big
[2020-09-14 11:40:39 UTC]   32KiB small
start
localhost [REQUEST s3.HeadObject] 11:40:44.787
localhost HEAD /test/big
localhost Proto: HTTP/1.1
localhost Host: localhost:9000
localhost Authorization: AWS4-HMAC-SHA256 Credential=admin/20200914/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=66434d1cbabac31003fa2ab35789b49193c520db55d7806cc54cf2a70571ae6c
localhost Connection: close
localhost Content-Length: 0
localhost User-Agent: aws-sdk-nodejs/2.751.0 linux/v10.22.0 promise
localhost X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
localhost X-Amz-Date: 20200914T114044Z
localhost
localhost [RESPONSE] [11:40:44.787] [ Duration 316µs  ↑ 88 B  ↓ 352 B ]
localhost 200 OK
localhost Server: MinIO/RELEASE.2020-09-10T22-02-45Z
localhost Vary: Origin
localhost Content-Type: application/octet-stream
localhost Last-Modified: Mon, 14 Sep 2020 11:40:39 GMT
localhost Content-Security-Policy: block-all-mixed-content
localhost ETag: "00000000000000000000000000000000-1"
localhost X-Amz-Request-Id: 1634A398EA77BDB8
localhost X-Xss-Protection: 1; mode=block
localhost Accept-Ranges: bytes
localhost Content-Length: 41943040
localhost
localhost

API: CopyObject(bucket=test, object=big)
Time: 11:40:45 UTC 09/14/2020
DeploymentID: d49c7684-4898-47c5-829a-8e34900ce0e8
RequestID: 1634A3992712908A
RemoteHost: 127.0.0.1
Host: localhost:9000
UserAgent: aws-sdk-nodejs/2.751.0 linux/v10.22.0 promise
Error: file not found
       5: cmd/fs-v1.go:625:cmd.(*FSObjects).CopyObject()
       4: cmd/object-handlers.go:1255:cmd.objectAPIHandlers.CopyObjectHandler()
       3: cmd/http-tracer.go:127:cmd.Trace()
       2: cmd/handler-utils.go:354:cmd.httpTraceAll.func1()
       1: net/http/server.go:2041:http.HandlerFunc.ServeHTTP()
localhost [REQUEST s3.CopyObject] 11:40:45.804
localhost PUT /test/big
localhost Proto: HTTP/1.1
localhost Host: localhost:9000
localhost Authorization: AWS4-HMAC-SHA256 Credential=admin/20200914/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-copy-source;x-amz-date;x-amz-meta-id1;x-amz-metadata-directive, Signature=750aa4047c0bc503cc70b0ca49591d7cabc8c88ac1777ba6abcbef2e27a020fb
localhost Content-Length: 0
localhost Content-Type: application/octet-stream
localhost User-Agent: aws-sdk-nodejs/2.751.0 linux/v10.22.0 promise
localhost X-Amz-Date: 20200914T114045Z
localhost X-Amz-Meta-Id1: Test
localhost X-Amz-Metadata-Directive: REPLACE
localhost Connection: close
localhost X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
localhost X-Amz-Copy-Source: test/big
localhost
localhost [RESPONSE] [11:40:45.804] [ Duration 430µs  ↑ 159 B  ↓ 550 B ]
localhost 404 Not Found
localhost Server: MinIO/RELEASE.2020-09-10T22-02-45Z
localhost Vary: Origin
localhost X-Amz-Request-Id: 1634A3992712908A
localhost X-Xss-Protection: 1; mode=block
localhost Accept-Ranges: bytes
localhost Content-Length: 293
localhost Content-Security-Policy: block-all-mixed-content
localhost Content-Type: application/xml
localhost <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>big</Key><BucketName>test</BucketName><Resource>/test/big</Resource><RequestId>1634A3992712908A</RequestId><HostId>d49c7684-4898-47c5-829a-8e34900ce0e8</HostId></Error>
localhost
{ NoSuchKey: The specified key does not exist.
    at Request.extractError (/home/node_modules/aws-sdk/lib/services/s3.js:837:35)
    at Request.callListeners (/home/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/home/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/home/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/home/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/home/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /home/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/home/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/home/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/home/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
  message: 'The specified key does not exist.',
  code: 'NoSuchKey',
  region: null,
  time: 2020-09-14T11:40:45.809Z,
  requestId: '1634A3992712908A',
  extendedRequestId: undefined,
  cfId: undefined,
  statusCode: 404,
  retryable: false,
  retryDelay: 49.06644666051376 }
```

Dockerfile:
```
from minio/minio:RELEASE.2020-09-10T22-02-45Z as minio
from node:10

RUN mkdir /CREDITS
COPY --from=minio /usr/bin/minio /usr/bin/minio
COPY --from=minio /CREDITS/ /CREDITS/
COPY --from=minio /usr/bin/docker-entrypoint.sh /usr/bin/docker-entrypoint.sh

ENV MINIO_ACCESS_KEY=admin
ENV MINIO_SECRET_KEY=admintest
ENV npm_config_loglevel=error

WORKDIR /home
RUN head -c 32k </dev/urandom | base64 - | head -c 32k > small
RUN head -c 40M </dev/urandom | base64 - | head -c 40M > big
RUN npm install aws-sdk
RUN wget -q https://dl.min.io/client/mc/release/linux-amd64/mc
RUN chmod +x mc
RUN mv mc /usr/bin/mc
COPY test.js /home/test.js
COPY startup.sh /home/startup.sh
RUN chmod +x /home/startup.sh

CMD ["/home/startup.sh"]
```

startup.sh:
```
#/bin/sh
mkdir -p /data/test 
/usr/bin/docker-entrypoint.sh minio server /data & 
sleep 5
mc alias set minio http://localhost:9000 $MINIO_ACCESS_KEY $MINIO_SECRET_KEY 
mc admin trace minio --debug -v & 
cp /home/small /data/test/
cp /home/big /data/test/
ls -l /data/test
mc ls minio/test
sleep 5
node /home/test.js
```

test.js:
```
const fs = require('fs');
const file = 'small';
const file2 = 'big';
const bucket = 'test';
const http = require('http');
const AWS = require("aws-sdk");
const s3Client = new AWS.S3({
    accessKeyId: 'admin',
    secretAccessKey: 'admintest',
    endpoint: 'http://localhost:9000',
    sslEnabled: false,
    s3ForcePathStyle: true,
    httpOptions: {
        agent: new http.Agent({})
    }
});

function Sleep(milliseconds) {
    return new Promise(resolve => setTimeout(resolve, milliseconds));
}

async function head() {
    let data = await s3Client.headObject({
        Bucket: bucket,
        Key: file2
    }).promise();

    if (!data) {
        console.log('head failed for ' + file2 + ' with:', err);
        return;
    }
    let origMeta = data['Metadata'];
    let myMeta = { id1: 'Test' };
    let fullMeta = Object.assign(origMeta, myMeta);
    await Sleep(1000);
    await copy(bucket, file2, fullMeta);
    return;
}

async function copy(bucket, key, metadata) {
    try {
        let data = await s3Client.copyObject({
            Bucket: bucket,
            Key: key,
            CopySource: bucket + '/' + key,
            Metadata: metadata,
            MetadataDirective: 'REPLACE'
        }).promise();

        if (!data) {
            console.log('copy failed for ' + bucket + '/' + key + ' with:', d);
            return;
        }
    } catch (err) {
        console.log(err);
    }
}

console.log('start');
head()    
```

## Context

We are using Noobaa to upload files to MinIO (deployed as standalone in an OKD 4.5 cluster) and after every upload we trigger a nodejs function which performs a headObject to retrieve the metadata. To the original metadata we add an id and use copyObject with the replace metadata directive to add this id to the object. This fails from time to time with a 404 Not Found, even though the headObject before succeeds. I was not able to reproduce the complete setup with the upload from the aws nodejs sdk in a container and getting the same error behavior.

Also the root cause seems to be a race condition, because i can easily reproduce the error when using port forwarding when i have low latency and can not reproduce it with higher latency. I'll provide logs for this race condition tommorow.

## Your Environment

* Version used (`minio --version`): RELEASE.2020-09-10T22-02-45Z
* Server setup and configuration: OKD 4.5 cluster
* Operating System and version (`uname -a`): Linux minio-5d5bf4bbb4-57p5v 5.6.19-300.fc32.x86_64 #1 SMP Wed Jun 17 16:10:48 UTC 2020 x86_64 Linux

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: allow CopyObject/PutObjecTags on pre-existing content

## Motivation and Context
fixes #10475

## How to test this PR?
As per reproducer in #10475

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#de93d253-37e4-438b-9512-e82188a085e9) |
| mint-gateway-azure.sh | :x: [more...](#7a327adc-1088-4952-b9c6-451565380ff9) |
<a name="7a327adc-1088-4952-b9c6-451565380ff9"></a>
### 10485-de909fc/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31462
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 319fff7c4d43:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 40 seconds
(4/15) Running aws-sdk-ruby tests ... done in 17 seconds
(5/15) Running awscli tests ... FAILED in 46 seconds
{
  "name": "awscli",
  "duration": 3911,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:31462 s3api copy-object --bucket awscli-mint-test-bucket-25066 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-25066/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected a861ac626e0bee4421a50737c581233c, got 0x8D85944ABA3CC73-1"
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 3 minutes and 12 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 30 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 40 seconds
{
  "args": {
    "bucketName": "minio-go-test-c0bok6rrlb59qms3",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 751,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 7 minutes and 47 seconds
(9/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 76,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:678:10) at endReadableNT (_stream_readable.js:1244:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(9/15) Running minio-py tests ... done in 12 minutes and 34 seconds
(10/15) Running s3cmd tests ... done in 2 minutes and 8 seconds
(11/15) Running s3select tests ... done in 34 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
<a name="de93d253-37e4-438b-9512-e82188a085e9"></a>
### 10485-de909fc/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30125
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 053bf5dbe9d6:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 8 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 51 seconds
(8/15) Running minio-dotnet tests ... done in 33 seconds
(9/15) Running minio-go tests ... FAILED in 47 seconds
{
  "args": {
    "bucketName": "minio-go-test-fmc2aoqchj5sk6pn",
    "objectName": "nvvmzqhedv5qdcf2evjgrrahsqrih5",
    "sse": {}
  },
  "duration": 42,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[SSE-S3]",
  "duration": 13,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-2qra78e, objectName = minio-java-test-1l6n5tj, resource = /minio-java-test-2qra78e/minio-java-test-1l6n5tj, requestId = 1634E2D07017FD51, hostId = 9075a18b-0596-4392-a554-ce7f620ac222)\nrequest={method=PUT, url=http://minio-c3.minio.io:30125/minio-java-test-2qra78e/minio-java-test-1l6n5tj, headers=X-Amz-Server-Side-Encryption: AES256\nContent-Type: application/javascript\nHost: minio-c3.minio.io:30125\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/7.1.0\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20200915T065912Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200915/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-server-side-encryption, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 402\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-09-15T06-44-17Z\nVary: Origin\nX-Amz-Request-Id: 1634E2D07017FD51\nX-Xss-Protection: 1; mode=block\nDate: Tue, 15 Sep 2020 06:59:12 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1135), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.testPutObject(FunctionalTest.java:740), FunctionalTest.putObject_test(FunctionalTest.java:887), FunctionalTest.runTests(FunctionalTest.java:3789), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(9/15) Running minio-js tests ... done in 39 seconds
(10/15) Running minio-py tests ... done in 1 minutes and 58 seconds
(11/15) Running s3cmd tests ... done in 1 minutes and 3 seconds
(12/15) Running s3select tests ... done in 3 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10485-de909fc

**Review by klauspost — APPROVED** [review]
LGTM

File: cmd/fs-v1.go [hunk 1]
```diff
@@ -622,8 +623,15 @@ func (fs *FSObjects) CopyObject(ctx context.Context, srcBucket, srcObject, dstBu
 		fsMetaPath := pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix, srcBucket, srcObject, fs.metaJSONFile)
 		wlk, err := fs.rwPool.Write(fsMetaPath)
 		if err != nil {
-			logger.LogIf(ctx, err)
-			return oi, toObjectErr(err, srcBucket, srcObject)
+			if !errors.Is(err, errFileNotFound) {
+				logger.LogIf(ctx, err)
+				return oi, toObjectErr(err, srcBucket, srcObject)
+			}
+			wlk, err = fs.rwPool.Create(fsMetaPath)
```
Comment (vadmeste) [inline cmd/fs-v1.go:630]: I am not able to understand the logic here, could you explain more ?

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/fs-v1.go (hunk 1 above)
Comment (klauspost) [reply cmd/fs-v1.go:630]: I believe this will cause the metadata to be replaced after `fsMeta.ReadFrom(ctx, wlk)` with the default data further down.

**Pull request merged**